modes.go

```diff
@@ -413,11 +408,29 @@ func DoReadsFromTable(table string, session *gocql.Session, threadResult *result
 		request = fmt.Sprintf("SELECT %s FROM %s.%s WHERE pk = ? AND ck >= ? %s LIMIT %d", selectFields, keyspaceName, table, orderBy, rowsPerRequest)
 	}
 
-	query := session.Query(request)
+	return session.Query(request)
+}
+
+func DoReadsFromTable(table string, session *gocql.Session, threadResult *results.TestThreadResult, workload WorkloadGenerator, rateLimiter RateLimiter) {
+	var queries []*gocql.Query
+	queryIdx := 0
+	queryIdxMax := len(selectOrderByParsed) - 1
+
+	for _, orderBy := range selectOrderByParsed {
+		queries = append(queries, BuildReadQuery(keyspaceName, table, orderBy, session))
+	}
 
 	RunTest(threadResult, workload, rateLimiter, func(rb *results.TestThreadResult) (error, time.Duration) {
 		pk := workload.NextPartitionKey()
 
+		if queryIdx >= queryIdxMax {
+			queryIdx = 0
+		} else {
+			queryIdx++
+		}
```

Comment:
I find this code confusing. It starts to execute queries from the query under idx equal to 1 not 0. It would be less confusing to do the increment after `query := queries[queryIdx]`.

Reply:
Yes it does, fixed.